tpls/desul/include/desul/atomics/Lock_Array_SYCL.hpp

```diff
@@ -128,13 +128,26 @@ inline void unlock_address_sycl(void* ptr, MemoryScopeNode) {
   lock_node_ref.exchange(0);
 }
 #else
+
+template <typename /*AlwaysInt*/ = int>
```

Comment:
If I get this right, this is the case where locks are not supported right? May want a comment here.

Reply:
Maybe so, except that this is the sync'ing back up with Desul, I am not technically proposing any change to the SYCL backend.  See the description.

Reply:
Added a comment